tests/test_mirror.py

```diff
@@ -6,228 +6,762 @@
 
-        # assert the ACL rule is created
-        atbl = swsscommon.Table(adb, "ASIC_STATE:SAI_OBJECT_TYPE_ACL_ENTRY")
-        keys = atbl.getKeys()
-        acl_entries = [k for k in keys if k not in dvs.asicdb.default_acl_entries]
-        assert len(acl_entries) == 1
+        # check asic database
+        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_MIRROR_SESSION")
+        assert len(tbl.getKeys()) == 1
 
-        # assert the ACL rule content is correct
-        (status, fvs) = atbl.get(acl_entries[0])
+        (status, fvs) = tbl.get(tbl.getKeys()[0])
         assert status == True
-        assert len(fvs) == 6
         for fv in fvs:
-            if fv[0] == "SAI_ACL_ENTRY_ATTR_TABLE_ID":
-                assert fv[1] == test_acl_table_id
-            elif fv[0] == "SAI_ACL_ENTRY_ATTR_ADMIN_STATE":
-                assert fv[1] == "true"
-            elif fv[0] == "SAI_ACL_ENTRY_ATTR_PRIORITY":
-                assert fv[1] == "1000"
-            elif fv[0] == "SAI_ACL_ENTRY_ATTR_ACTION_COUNTER":
-                assert True
-            elif fv[0] == "SAI_ACL_ENTRY_ATTR_FIELD_DSCP":
-                assert fv[1] == "16&mask:0x10"
-            elif fv[0] == "SAI_ACL_ENTRY_ATTR_ACTION_MIRROR_INGRESS":
-                assert fv[1] == "1:" + test_mirror_session_id
-            else:
-                assert False
+            if fv[0] == "SAI_MIRROR_SESSION_ATTR_MONITOR_PORT":
+                assert dvs.asicdb.portoidmap[fv[1]] == "Ethernet88"
+            elif fv[0] == "SAI_MIRROR_SESSION_ATTR_DST_MAC_ADDRESS":
+                assert fv[1] == "88:88:88:88:88:88"
 
-        # remove the ACL rule
-        tbl._del("EVERFLOW_TABLE|EVERFLOW_DSCP_TEST_RULE_2")
-        time.sleep(1)
+        # remove neighbor
+        self.remove_neighbor("PortChannel008", "11.11.11.11")
+        assert self.get_mirror_session_status(session) == "inactive"
+
+        # remove ip address
+        self.remove_ip_address("PortChannel008", "11.11.11.0/24")
+        assert self.get_mirror_session_status(session) == "inactive"
+
+        # bring down port channel and port channel member
+        self.set_interface_status("PortChannel008", "down")
+        self.set_interface_status("Ethernet88", "down")
 
-        # assert the ACL rule is removed
-        (status, fvs) = atbl.get(acl_entries[0])
-        assert status == False
+        # remove port channel member; remove port channel
+        self.remove_port_channel_member("008", "Ethernet88")
+        self.remove_port_channel(dvs, "008")
 
-    def test_AclMirrorTableDeletion(self, dvs):
-        adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
-        cdb = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+        # remove mirror session
+        self.remove_mirror_session(session)
 
-        tbl = swsscommon.Table(cdb, "ACL_TABLE")
-        # remove the ACL table
-        tbl._del("EVERFLOW_TABLE")
+
+    def test_MirrorDestMoveVlan(self, dvs):
```

Comment:
Do you have test case for MirrorDestMoveNonVlan to test move from vlan to non-vlan case?

Reply:
this test covers both cases: vlan to non-vlan and vice versa